adds our shuttles to tg's shuttle manipulator

dwasint opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds all our shuttles to the tg shuttle manipulator dynamically. Ships will now load and show up as shuttle in the shuttle manager unless the mobile docking port is named in the map files
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:

/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

dwasint committed:
adds all the voidcrew ships to shuttle manipulator

TODO: add spawning via shuttle menu

dwasint committed:
fixes

dwasint committed:
ag

dwasint committed:
merge upstream

JohnFulpWillard commented on voidcrew/code/datums/shuttles.dm:
```diff
@@ -34,3 +34,14 @@
+	mappath = "[prefix][shuttle_id].dmm"
+	. = ..()
+
+/datum/map_template/shuttle/voidcrew
```
please don't put /New above it

dwasint replied:
I think i fixed this? Not to sure what you meant

JohnFulpWillard commented on voidcrew/code/datums/shuttles.dm:
```diff
@@ -34,3 +34,14 @@
+/datum/map_template/shuttle/voidcrew/New()
+	shuttle_id = "[suffix]"
+	mappath = "[prefix][shuttle_id].dmm"
+	. = ..()
```
```suggestion
	. = ..()
	shuttle_id = "[suffix]"
	mappath = "[prefix][shuttle_id].dmm"
```

JohnFulpWillard commented on voidcrew/code/controllers/subsystem/mapping.dm:
```diff
@@ -31,3 +33,91 @@
+				nt_ship_list[shuttle.name] = shuttle
+			if("SYN-C")
+				syn_ship_list[shuttle.name] = shuttle
+#undef CHECK_STRING_EXISTS
```
```suggestion

#undef CHECK_STRING_EXISTS
```

JohnFulpWillard commented on code/controllers/subsystem/shuttle.dm:
```diff
@@ -659,14 +659,19 @@ SUBSYSTEM_DEF(shuttle)
 /datum/controller/subsystem/shuttle/proc/is_in_shuttle_bounds(atom/A)
 	var/area/current = get_area(A)
-	if(istype(current, /area/shuttle) && !istype(current, /area/shuttle/transit))
+	if((istype(current, /area/shuttle) && !istype(current, /area/shuttle/transit)) || istype(current, /area/ship))
```
this file can be made modular if you make a subtype of it that isn't actually a subtype (just remove /proc) and call parent.

dwasint replied:
I think this change is actually just not needed, dennok's pr showed you just need to modify the ship itself to work

dwasint committed:
guh

dwasint committed:
Reverts unneeded changes

dwasint committed:
fixes bead?

dwasint committed:
oops

dwasint committed:
fixes all shuttles hopefully minus a few that are totalled right now

Lamia and blackbeard still need fixes before anything as lamia is missing all its walls and blackbeard is missing some suits

dwasint committed:
maybe fixed?

dwasint committed:
Revert "maybe fixed?"

This reverts commit 59a99120752a9f87f857fc99c9d2ca115be80c92.

dwasint committed:
Fixes almost everything

dwasint committed:
twinkleshine fixed

dwasint committed:
Merge branch 'tg-rebase' into tg-shuttle-manipulator

dwasint committed:
Merges from upstream

dwasint committed:
Delete .merge_file_a15484

dwasint committed:
Delete .merge_file_a29320

dwasint committed:
Delete .merge_file_b15484

dwasint committed:
Delete .merge_file_b29320

dwasint committed:
Delete .merge_file_c15484

dwasint committed:
Delete .merge_file_c29320

dwasint committed:
fixes bead

dwasint committed:
Merge branch 'tg-shuttle-manipulator' of https://github.com/dwasint/Voidcrew-LRP into tg-shuttle-manipulator

dwasint committed:
guh

dwasint committed:
fixes the last of the ships

dwasint committed:
reverts un-needed changes
